nix/supervisord-cluster/supervisor.nix

```diff
@@ -23,12 +23,15 @@ let
       extra = {
         services.cardano-node = {
           enable = true;
-          ## topology wtf?
-          inherit (envConfig) operationalCertificate kesKey vrfKey topology nodeConfig nodeConfigFile port dbPrefix socketPath;
+          inherit (envConfig) topology nodeConfig nodeConfigFile port dbPrefix socketPath;
           inherit stateDir;
-        } // lib.optionalAttrs useCabalRun
-          { executable = "cabal run exe:cardano-node --";
-          };
+        }
+        // lib.optionalAttrs (__hasAttr "vrfKey" envConfig)
+        { inherit (envConfig) operationalCertificate kesKey vrfKey;
+        }
+        // lib.optionalAttrs useCabalRun
+        { executable = "cabal run exe:cardano-node --";
```

Comment:
Is it not the case that cabal and nix are somehow orthogonal ?
Why is the executable called with 'cabal run exe:cardano-node' ?
I would have expected that the executable is defined in a nix-derivation ?

Reply:
First, that's not the new part here -- it's just shuffling code around.  This code was introduced in the previous PR.

Second, the idea is that we're reusing the `cardano-node` service definition here -- precisely to be able to use the Nix infrastructure to give use all the dependencies, but _also_ use `cabal build/run` to get the "last mile" -- the node itself.